Move TypeLoadException to shared

EgorBo opened the pull request:
Or it's better to if-def CoreCLR specific code for changes like this?

jkotas commented on src/System.Private.CoreLib/src/System/TypeLoadException.CoreCLR.cs:
```diff
@@ -0,0 +1,47 @@
+
+namespace System
+{
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
```
This should be in the share part.

EgorBo replied:
Both Mono and CoreRT don't have it, should I still move the attribute there?

jkotas replied:
CoreRT has it: https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/TypeLoadException.cs#L10

And all exceptions in shared have it too.

EgorBo replied:
ah indeed

jkotas commented:
Fix the field names to follow the coding conventions?

jkotas commented:
Can `ResourceId` be private?

jkotas commented:
> Or it's better to if-def CoreCLR specific code for changes like this?

How much of the the logic moved to shared is Mono going to take advantage of?

EgorBo commented:
> > Or it's better to if-def CoreCLR specific code for changes like this?
> 
> How much of the the logic moved to shared is Mono going to take advantage of?

um.. not sure I understand the question, the whole file except the qcall

BTW, addressed the feedback

jkotas commented:
> not sure I understand the question

E.g. does Mono use and set the `_resourceId` field?

EgorBo commented:
@jkotas the old mono did used `ResourceId` property, but I guess since coreclr/corefx don't use it anywhere we also don't need it, @marek-safar thoughts?

marek-safar commented on src/System.Private.CoreLib/shared/System/TypeLoadException.cs:
```diff
@@ -106,30 +68,27 @@ public string TypeName
+            _messageArg = info.GetString("TypeLoadMessageArg");
+            _resourceId = info.GetInt32("TypeLoadResourceID");
         }
 
```
private TypeLoadException ctor is CoreCLR specific

marek-safar commented:
> E.g. does Mono use and set the _resourceId field?

it does not but I expect we need to serialize it for compatibility reasons. Also not sure what the globalization story will be going forward so we might need to switch to using it.

jkotas commented on src/System.Private.CoreLib/shared/System/TypeLoadException.cs:
```diff
@@ -0,0 +1,76 @@
+**
+**
+**
+** Purpose: The exception class for type loading failures.
```
Nit: We have been deleting useless comments like this when moving stuff to shared.
